### DISCUSSION TITLE
Is there a docker for building image from source?

### DISCUSSION THREAD

**Discussion by feiyax** [Joshua-Riek/ubuntu-rockchip#1027] [Q&A]
Looking at building from source but i hope to do so in a docker container.

Is there already a container or Dockerfile somewhere that I can grab and use as builder? 

If not, are there details on what the build environment should be (so i can set up my own container)? (Such as what distro and what packages).

Thanks.

**Comment by Joshua-Riek** [comment]
no

**Reply by feiyax** [reply]
What'd be a recommended build environment, in terms of distro and packages? 

Say if i'm to set up a VM to build, what distro (assuming it's a debian/ubuntu one) and what `apt install <pkgs>` should i use? Thanks.

**Comment by Joshua-Riek** [comment]
Docker will not work when building a Ubuntu 24.04 rootfs. When using the Desktop rootfs it will be all fucked up.

**Comment by peterfortuin** [comment]
If docker is not a option (not really understand why it is not a option), can you tell what kind of environment I do need to build a image? Just a Linux / Ubuntu distribution? Intel / Arm? Does it matter? What kind of packages do I need to install?

**Reply by ethanc8** [reply]
This might be helpful, but I have not verified it: https://github.com/Joshua-Riek/ubuntu-rockchip/discussions/1076. Additionally, it works to build it on GitHub Actions (I believe this is how the official images are built) - see https://github.com/Joshua-Riek/ubuntu-rockchip/tree/main/.github/workflows.
